File: ByteDance/21. 合并两个有序链表.go
package main

// 21. 合并两个有序链表

// 将两个升序链表合并为一个新的 升序 链表并返回。新链表是通过拼接给定的两个链表的所有节点组成的。

func main() {

}

func mergeTwoLists(l1 *ListNode, l2 *ListNode) *ListNode {
	if l1 == nil {
		return l2
	} else if l2 == nil {
		return l1
	}

	var merge *ListNode
	if l1.Val > l2.Val {
		merge = l2
		merge.Next = mergeTwoLists(l1, l2.Next)
	} else {
		merge = l1
		merge.Next = mergeTwoLists(l1.Next, l2)
	}

	return merge
}

File: C++/21.合并两个有序链表.cpp

// 21. 合并两个有序链表
//
// 将两个升序链表合并为一个新的 升序 链表并返回。新链表是通过拼接给定的两个链表的所有节点组成的。

#include "header.h"

class Solution {
public:
    // 同offer 25
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2)
    {
        if (list1 == nullptr) {
            return list2;
        } else if (list2 == nullptr) {
            return list1;
        } else if (list1->val < list2->val) {
            list1->next = mergeTwoLists(list1->next, list2);
            return list1;
        } else {
            list2->next = mergeTwoLists(list1, list2->next);
            return list2;
        }
    }
};
